### PR TITLE
Remove counter mutations from controls_for_counter blocks

### PR DESCRIPTION
**Pull request by smusoke**
**What:** 
- Remove `counter` variable hardcoding within `controls_for_counter` blockly blocks. The `levels#update_blocks` endpoint will remove the mutations stored on the blocks before saving the xml to the level.
- Oneoff script to remove all mutations from blocks in existing level files. This doesn't include `authored_hints` or other fields within the level config

**Why**:  Once a mutation is stored on a block this is essentially used as state. This results in the variable's string being permanently set when saved to the level, and not allowing for translation.

## Links

- jira ticket: [FND-1681](https://codedotorg.atlassian.net/browse/FND-1681)

## Testing story
https://adhoc-fnd-1681-counter-hardcoding-studio.cdn-code.org/s/express-2017/lessons/26/levels/2
- Ran one off locally, reseeded the database, tested handful of levels locally
- Tested editing the start_blocks, toolbox, and solution in levelbuilder mode on the adhoc
- No regression with existing mutation purpose
https://codedotorg.slack.com/archives/C1B3PNDL7/p1630425480011200?thread_ts=1630357332.009200&cid=C1B3PNDL7

Deutsch saved project
Screen Shot 2021-09-17 at 12 07 38 PM

Opened and working in Espanol (counter variable hardcoded in original language is expected)

Screen Shot 2021-09-17 at 12 11 09 PM

## Follow-up work

Reseed levels in database? (Not sure if this is done every DTP)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/levels/blockly.rb [hunk 1]
```diff
@@ -226,6 +226,14 @@ def self.convert_category_to_toolbox(xml_string)
     xml.serialize(save_with: XML_OPTIONS).delete("\n").strip
   end
 
+  # "counter" mutations should not be stored because it results in the language being hardcoded
```
Comment (ajpal) [inline dashboard/app/models/levels/blockly.rb]: Can you elaborate in this comment that the mutation *should* be stored for student code, it just shouldn't be stored in the level configuration XML (start/toolbox/solution blocks)

File: bin/oneoff/remove_controls_for_counter_mutations.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    updated_levels += 1
+  rescue Exception => e
+    # print filename for better debugging
+    new_e = Exception.new("in level: #{path}: #{e.message}")
+    new_e.set_backtrace(e.backtrace)
+    raise new_e
+  end
+
+  puts "#{skipped_levels} level files didn't need updating"
+  puts "#{updated_levels} level files that had a controls_for_counter mutation removed"
```
Comment (ajpal) [inline bin/oneoff/remove_controls_for_counter_mutations.rb:55]: I didn't look at all the .level files, but is it possible there were blocks other than `controls_for_counter` that had a `"counter"` mutation set that would have been inadvertently stripped out?

Comment (smusoke) [reply bin/oneoff/remove_controls_for_counter_mutations.rb:55]: This is a good point. It makes sense to me [to update the xpaths to make sure the blocks being removed are descendants of a `controls_for_counter` block](https://github.com/code-dot-org/code-dot-org/commit/4d7a7a1bcf2c1932e5268b40eca740693c9c88bd#diff-b4a749007acbcf28c0dd7c4cbe6d3dc12aa0768ca156ed6777c8fa91b6ae3411R37). I also ran the one off again with this update.

File: dashboard/app/models/levels/blockly.rb (hunk 1 above)
Comment (smusoke) [reply dashboard/app/models/levels/blockly.rb]: Yes, definitely should have as much clarity as possible here.

**Review by ajpal — APPROVED** [review]
lgtm! awesome work on this :)

**Pull request merged**